docs/guides/src/main/server/outgoinghttp.adoc

```diff
@@ -0,0 +1,147 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+
+<@tmpl.guide
+title="Configure outgoing HTTP requests"
+summary="How to configure Keycloak's HTTP client for outgoing requests."
+includedOptions="">
+
+Keycloak often needs to make requests to the applications and services it secures. Keycloak manages these outgoing connections using an HTTP client. This guide shows how to configure the client, it's connection pool, proxy environment settings, timeouts and more.
+
+== Usage
+To configure Keycloak's outgoing Http client, use:
+
+<@kc.start parameters="--spi-connections-http-client-default-<configurationoption>=<value>"/>
+
+Please see <<Outgoing Http Client Configuration Options>> for the available configuration options.
+
+== Outgoing Http Client Configuration Options
+
+The Http client Keycloak uses for outgoing communication is highly configurable. The available options are:
+
+*establish-connection-timeout-millis*::
+Maximum time in milliseconds until a connection approach times out. Default: Not set.
+
+*socket-timeout-millis*::
+Maximum time of inactivity between two data packets until a socket connection times out, in milliseconds. Default: 5000ms
+
+*connection-pool-size*::
+Size of the connection pool for outgoing connections. Default: 128.
+
+*max-pooled-per-route*::
+How many connections can be pooled per host. Default: 64.
+
+*connection-ttl-millis*::
+Maximum connection time to live in milliseconds. Default: Not set.
+
+*max-connection-idle-time-millis*::
+Maximum time an idle connection stays in the connection pool, in milliseconds. Idle connections will be removed from the pool by a background cleaner thread. Set to -1 to disable this check. Default: 900000.
+
+*disable-cookies*::
+Enable or disable caching of cookies. Default: true.
```

Comment:
Wonder why we have cookies enabled to be honest, that should probably not be enabled as we're not doing calls from a user-agent and shouldn't need/support cookies

Reply:
I don't know why, I just looked at https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/connections/httpclient/DefaultHttpClientFactory.java#L143-L157 for the defaults and added them. @pedroigor wdyt, change this?

Reply:
re-reading this: actually, cookies are disabled by default bc. this is not "enable-cookies=true", but "disable-cookies=true". 🙈